mat/matrix.go

```diff
@@ -345,7 +343,7 @@ func Dot(a, b Vector) float64 {
 	}
```

Comment:
In the condition above, we are now checking twice.

Reply:
Done.

Reply:
No, sorry. That was not something to change. It's an unfortunate consequence of the checks in blas64 now. We have to check twice. We promise to `panic(ErrShape)` in mat for this situation. As it stands now, we don't.

Reply:
Good catch. Reverted.